Maple2.Server.Game/PacketHandlers/MasteryHandler.cs

```diff
@@ -32,12 +32,52 @@ private enum Command : byte {
     public override void Handle(GameSession session, IByteReader packet) {
         var command = packet.Read<Command>();
         switch (command) {
+            case Command.Reward:
+                HandleReward(session, packet);
+                break;
             case Command.Craft:
                 HandleCraft(session, packet);
                 break;
         }
     }
 
+    private void HandleReward(GameSession session, IByteReader packet) {
+        int rewardBoxDetails = packet.ReadInt();
+        var type = (MasteryType) (rewardBoxDetails / 1000);
+        int level = rewardBoxDetails % 100;
+
+        if (session.Player.Value.Unlock.MasteryRewardsClaimed.TryGetValue(rewardBoxDetails, out bool isClaimed) && isClaimed) {
```

Comment:
This check seems wrong, if the id is not in `MasteryRewardsClaimed`, shouldn't it be considered valid? Perhaps you instead mean: `if (!TryGetValue(...) || isClaimed)`?